external/asf4-drivers/qtouch/qtouch.c

```diff
@@ -604,17 +608,25 @@ void qtouch_process_scroller_positions(void)
 
         // Calculate scroller position using a moving average
         // The moving average wieghts previous N readings twice current reading
+        uint16_t estimated_scroller_position;
         if (sum >= DEF_SCROLLER_DET_THRESHOLD) {
             uint16_t moving_average_cummulative_weight = 0;
-            scroller_position[scroller] = 0;
+            estimated_scroller_position = 0;
             for (j = 0; j < DEF_SCROLLER_NUM_PREV_POS; j++) {
                 if (scroller_previous_position[scroller][j] != DEF_SCROLLER_OFF) {
                     moving_average_cummulative_weight += 2;
-                    scroller_position[scroller] += scroller_previous_position[scroller][j] * 2;
+                    estimated_scroller_position += scroller_previous_position[scroller][j] * 2;
                 }
             }
-            scroller_position[scroller] += scaled_value;// Most recent signal is half weight of others to help avoid bounce when finger is released
-            scroller_position[scroller] = scroller_position[scroller] / (moving_average_cummulative_weight + 1);
+            estimated_scroller_position += scaled_value;// Most recent signal is half weight of others to help avoid bounce when finger is released
+            estimated_scroller_position = estimated_scroller_position / (moving_average_cummulative_weight + 1);
+        } else {
+            estimated_scroller_position = 0;
+        }
+
+        // Position-independent pressure estimation.
+        if (sum >= DEF_SCROLLER_DET_THRESHOLD) {
```

Comment:
can be part of above conditional (which is the same conditional) to reduce lines of code